### PR TITLE
FIX: Justify content in viewer

### PR DESCRIPTION
**Pull request by ShamaySapir**

### CODE REVIEW

File: packages/viewer/web/src/utils/convertContentState.jsx [hunk 1]
```diff
@@ -106,7 +122,7 @@ const getBlocks = (mergedStyles, textDirection, context, addAnchorsPrefix) => {
             style={blockDataToStyle(blockProps.data[i])}
             key={blockProps.keys[i]}
           >
-            {_child}
+            <span className={shouldJustify && styles.justifiedText}>{_child}</span>
```
Comment (shaulgo) [inline packages/viewer/web/src/utils/convertContentState.jsx:125]: this changes the dom structure. I rather we don't change it, unless we have to.

File: packages/viewer/web/src/utils/convertContentState.jsx [hunk 2]
```diff
@@ -29,6 +30,15 @@ const isEmptyBlock = ([_, data]) => data && data.length === 0; //eslint-disable-
...
   contentState.blocks.find(block => block.key === key).depth || 0;
 
+// eslint-disable-next-line no-unused-vars
+const hasText = ([_, data]) => {
+  let characters = false;
+  if (data[0] && data[0].replace(/\s/g, '').length !== 0) {
+    characters = true;
+  }
+  return characters;
+};
```
Comment (shaulgo) [inline packages/viewer/web/src/utils/convertContentState.jsx:40]:
```suggestion
const hasText = ([_, data]) => data[0]?.trim().length > 0
```

File: packages/viewer/web/statics/rich-content-viewer.scss [hunk 3]
```diff
@@ -123,6 +123,10 @@
   & > * {
     white-space: pre-wrap !important;
     text-align: justify !important;
+
+    &.justifiedText {
+      white-space: pre-line !important;
+    }
   }
```
Comment (shaulgo) [inline packages/viewer/web/statics/rich-content-viewer.scss:130]:
```suggestion
  }
  &.hasText {
    white-space: pre-wrap !important;
    & > * {
      white-space: pre-wrap !important;
    }
  }
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/viewer/web/src/utils/convertContentState.jsx (hunk 2 above)
Comment (shaulgo) [reply packages/viewer/web/src/utils/convertContentState.jsx:40]: or what about isEmptyBlock from below?

File: packages/viewer/web/src/utils/convertContentState.jsx (hunk 1 above)
Comment (ShamaySapir) [reply packages/viewer/web/src/utils/convertContentState.jsx:125]: fixed

File: packages/viewer/web/src/utils/convertContentState.jsx (hunk 2 above)
Comment (ShamaySapir) [reply packages/viewer/web/src/utils/convertContentState.jsx:40]: isEmptyBlock checks if there is no text at all, but it considers white spaces as not empty.

Comment (ShamaySapir) [reply packages/viewer/web/src/utils/convertContentState.jsx:40]: fixed with trim()

File: packages/viewer/web/statics/rich-content-viewer.scss (hunk 3 above)
Comment (ShamaySapir) [reply packages/viewer/web/statics/rich-content-viewer.scss:130]: fixed

File: packages/viewer/web/statics/rich-content-viewer.scss [hunk 4]
```diff
@@ -117,11 +117,8 @@
 }
 
 .justify {
```
Comment (shaulgo) [inline packages/viewer/web/statics/rich-content-viewer.scss:119]: .justify.hasText {

File: packages/viewer/web/statics/rich-content-viewer.scss [hunk 5]
```diff
@@ -117,11 +117,8 @@
...
-  white-space: pre-wrap !important;
-  text-align: justify !important;
-
-  & > * {
-    white-space: pre-wrap !important;
```
Comment (shaulgo) [inline packages/viewer/web/statics/rich-content-viewer.scss:124]: not sure we should remove this. We need to see why it was added.

Comment (ShamaySapir) [reply packages/viewer/web/statics/rich-content-viewer.scss:124]: fixed

File: e2e/cypress/support/commands.js [hunk 6]
```diff
@@ -43,8 +43,26 @@ const getUrl = (componentId, fixtureName = '', config = {}) => {
   })}`;
 };
 
-const run = (app, fixtureName, plugins) => {
-  cy.visit(getUrl(app, fixtureName, plugins)).then(contentWindow => {
+function setUserAgent(window, userAgent) {
+  if (window.navigator.__defineGetter__) {
+    window.navigator.__defineGetter__('userAgent', () => {
+      return userAgent;
+    });
```
Comment (shaulgo) [inline e2e/cypress/support/commands.js:50]:
```suggestion
    window.navigator.__defineGetter__('userAgent', () => userAgent);
```

File: packages/viewer/web/statics/rich-content-viewer.scss [hunk 7]
```diff
@@ -117,12 +117,12 @@
...
 
 .justify {
+
   white-space: pre-wrap !important;
   text-align: justify !important;
 
-  & > * {
-    white-space: pre-wrap !important;
-    text-align: justify !important;
+  &.hasTextSafariFirefox {
```
Comment (shaulgo) [inline packages/viewer/web/statics/rich-content-viewer.scss:124]:
```suggestion
  &.hasTextAndSafariOrFirefox {
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: e2e/cypress/support/commands.js (hunk 6 above)
Comment (ShamaySapir) [reply e2e/cypress/support/commands.js:50]: fixed

File: packages/viewer/web/statics/rich-content-viewer.scss (hunk 7 above)
Comment (ShamaySapir) [reply packages/viewer/web/statics/rich-content-viewer.scss:124]: fixed

File: e2e/cypress/support/commands.js [hunk 8]
```diff
@@ -43,12 +43,32 @@ const getUrl = (componentId, fixtureName = '', config = {}) => {
...
+        onBeforeLoad: contentWindow => {
+          isFirefox && setUserAgent(contentWindow, 'firefox');
+        },
+      }).then(contentWindow => {
+        disableTransitions();
+        findEditorElement();
+        contentWindow.richContentHideTooltips = true;
+      });
+    });
 };
```
Comment (shaulgo) [inline e2e/cypress/support/commands.js:72]:
```suggestion    
      cy.visit(getUrl(app, fixtureName, plugins), {
        onBeforeLoad: contentWindow => {
        if (Cypress.env('firefox'))
          setUserAgent(contentWindow, 'firefox');
        },
      }).then(contentWindow => {
        disableTransitions();
        findEditorElement();
        contentWindow.richContentHideTooltips = true;
      });

};
```

File: e2e/tests/rendering.e2e.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*global cy Cypress*/
-import { DEFAULT_DESKTOP_BROWSERS } from './settings';
+import { DEFAULT_DESKTOP_BROWSERS, FIREFOX_BROWSERS } from './settings';
```
Comment (shaulgo) [inline e2e/tests/rendering.e2e.js:2]:
```suggestion
import { DEFAULT_DESKTOP_BROWSERS, FIREFOX_BROWSER } from './settings';
```

File: e2e/tests/rendering.e2e.js [hunk 10]
```diff
@@ -27,6 +35,22 @@ describe('editor rendering', () => {
...
 
+  context('firefoxDesktop', () => {
+    before(function() {
+      eyesOpenerFirefox(this.test.parent.title);
+    });
+
+    beforeEach(() => {
+      cy.window().then(win => {
+        win.firefox = true;
+      });
```
Comment (shaulgo) [inline e2e/tests/rendering.e2e.js:46]:
```suggestion
```

File: e2e/tests/rendering.e2e.js [hunk 11]
```diff
@@ -27,6 +35,22 @@ describe('editor rendering', () => {
     testFixtures();
   });
 
+  context('firefoxDesktop', () => {
```
Comment (shaulgo) [inline e2e/tests/rendering.e2e.js:38]:
```suggestion
  context('firefoxDesktop', {env: {firefox: true}}, () => {
```

File: e2e/tests/settings.js [hunk 12]
```diff
@@ -47,4 +47,8 @@ export const fixturesToTestOnSeo = ['images'];
 
 export const DEFAULT_DESKTOP_BROWSERS = [{ width: 1440, height: 900, name: 'chrome' }];
 
+export const SAFARI_BROWSERS = [{ width: 1440, height: 900, name: 'safari' }];
+
+export const FIREFOX_BROWSERS = [{ width: 1440, height: 900, name: 'firefox' }];
```
Comment (shaulgo) [inline e2e/tests/settings.js:52]:
```suggestion
export const FIREFOX_BROWSER = [{ width: 1440, height: 900, name: 'firefox' }];
```

File: e2e/tests/text.e2e.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable max-len */
 /*global cy*/
 import { INLINE_TOOLBAR_BUTTONS } from '../cypress/dataHooks';
-import { DEFAULT_DESKTOP_BROWSERS } from './settings';
+import { DEFAULT_DESKTOP_BROWSERS, FIREFOX_BROWSERS } from './settings';
```
Comment (shaulgo) [inline e2e/tests/text.e2e.js:4]:
```suggestion
import { DEFAULT_DESKTOP_BROWSERS, FIREFOX_BROWSER } from './settings';
```

File: e2e/tests/text.e2e.js [hunk 14]
```diff
@@ -254,3 +254,49 @@ describe('text', () => {
     });
   });
 });
+
+describe('textFirefox', () => {
+  before(function() {
+    cy.eyesOpen({
+      appName: 'textFirefox',
+      testName: this.test.parent.title,
+      browser: FIREFOX_BROWSERS,
```
Comment (shaulgo) [inline e2e/tests/text.e2e.js:263]:
```suggestion
      browser: FIREFOX_BROWSER,
```

File: e2e/tests/text.e2e.js [hunk 15]
```diff
@@ -254,3 +254,49 @@ describe('text', () => {
...
+    });
+  });
+
+  beforeEach(() => {
+    cy.window().then(win => {
+      win.firefox = true;
+    });
```
Comment (shaulgo) [inline e2e/tests/text.e2e.js:270]:
```suggestion
```

File: e2e/tests/text.e2e.js [hunk 16]
```diff
@@ -254,3 +254,49 @@ describe('text', () => {
     });
   });
 });
+
+describe('textFirefox', () => {
```
Comment (shaulgo) [inline e2e/tests/text.e2e.js:258]:
```suggestion
describe('textFirefox', {env: {firefox: true}}, () => {
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: e2e/tests/settings.js (hunk 12 above)
Comment (ShamaySapir) [reply e2e/tests/settings.js:52]: fixed

File: e2e/tests/text.e2e.js (hunk 13 above)
Comment (ShamaySapir) [reply e2e/tests/text.e2e.js:4]: fixed

File: e2e/tests/text.e2e.js (hunk 14 above)
Comment (ShamaySapir) [reply e2e/tests/text.e2e.js:263]: fixed

File: e2e/tests/rendering.e2e.js (hunk 9 above)
Comment (ShamaySapir) [reply e2e/tests/rendering.e2e.js:2]: fixed

File: e2e/tests/rendering.e2e.js (hunk 10 above)
Comment (ShamaySapir) [reply e2e/tests/rendering.e2e.js:46]: fixed

File: e2e/tests/text.e2e.js (hunk 15 above)
Comment (ShamaySapir) [reply e2e/tests/text.e2e.js:270]: fixed

File: e2e/tests/text.e2e.js (hunk 16 above)
Comment (ShamaySapir) [reply e2e/tests/text.e2e.js:258]: fixed

File: e2e/tests/rendering.e2e.js (hunk 11 above)
Comment (ShamaySapir) [reply e2e/tests/rendering.e2e.js:38]: fixed

File: e2e/cypress/support/commands.js (hunk 8 above)
Comment (ShamaySapir) [reply e2e/cypress/support/commands.js:72]: fixed

**Review by shaulgo — APPROVED** [review]

**Pull request merged**